test/hummingbot/strategy/cross_exchange_market_making/test_cross_exchange_market_making_gateway.py

```diff
@@ -0,0 +1,1017 @@
+    def name(self):
+        return self._name
+
+    @property
+    def network_transaction_fee(self) -> TokenAmount:
+        return self._network_transaction_fee
+
+    @network_transaction_fee.setter
+    def network_transaction_fee(self, fee: TokenAmount):
+        self._network_transaction_fee = fee
+
+    @property
+    def connector_name(self):
+        return "uniswap"
+
+    async def get_quote_price(self, trading_pair: str, is_buy: bool, amount: Decimal) -> Decimal:
+        if is_buy:
+            return self._buy_prices[trading_pair]
+        else:
+            return self._sell_prices[trading_pair]
+
+    async def get_order_price(self, trading_pair: str, is_buy: bool, amount: Decimal) -> Decimal:
+        return await self.get_quote_price(trading_pair, is_buy, amount)
+
+    def set_prices(self, trading_pair, is_buy, price):
+        if is_buy:
+            self._buy_prices[trading_pair] = Decimal(str(price))
+        else:
+            self._sell_prices[trading_pair] = Decimal(str(price))
+
+    def set_balance(self, token, balance):
+        self._account_balances[token] = Decimal(str(balance))
+        self._account_available_balances[token] = Decimal(str(balance))
+
+    def buy(self, trading_pair: str, amount: Decimal, order_type: OrderType, price: Decimal):
+        return self.place_order(True, trading_pair, amount, price)
+
+    def sell(self, trading_pair: str, amount: Decimal, order_type: OrderType, price: Decimal):
+        return self.place_order(False, trading_pair, amount, price)
+
+    def place_order(self, is_buy: bool, trading_pair: str, amount: Decimal, price: Decimal):
+        side = "buy" if is_buy else "sell"
+        order_id = f"{side}-{trading_pair}-{get_tracking_nonce()}"
+        event_tag = MarketEvent.BuyOrderCreated if is_buy else MarketEvent.SellOrderCreated
+        event_class = BuyOrderCreatedEvent if is_buy else SellOrderCreatedEvent
+        self.trigger_event(event_tag,
+                           event_class(
+                               self.current_timestamp,
+                               OrderType.LIMIT,
+                               trading_pair,
+                               amount,
+                               price,
+                               order_id,
+                               self.current_timestamp))
+        return order_id
+
+    def get_taker_order_type(self):
+        return OrderType.LIMIT
+
+    def get_order_price_quantum(self, trading_pair: str, price: Decimal) -> Decimal:
+        return Decimal("0.01")
+
+    def get_order_size_quantum(self, trading_pair: str, order_size: Decimal) -> Decimal:
+        return Decimal("0.01")
+
+    def estimate_fee_pct(self, is_maker: bool):
+        return Decimal("0")
+
+    def ready(self):
+        return True
+
+    async def check_network(self) -> NetworkStatus:
+        return NetworkStatus.CONNECTED
+
+    async def cancel_outdated_orders(self, _: int) -> List:
+        return []
+
+
+@attr("stable")
```

Comment:
We are not using test attributes to run the suites, we should not add them.

Reply:
ok, I'll remove it also from the original XEMM tests

Reply:
✅